src/ucp/api/ucp_def.h

```diff
@@ -771,5 +772,22 @@ typedef struct ucp_ep_params {
  */
 #define UCP_ENTITY_NAME_MAX 32
 
+/**
+ * @ingroup UCP_TRANSPORT
+ * @brief Maximum number of transport/device name pairs
+ */
+#define UCP_MAX_TRANSPORTS 10
```

Comment:
IMO this limitation can be too small for future changes.
maybe user will allocate an array of "ucp_transport_entry_t" and provide its size to the query function?

Reply:
I changed the code so the caller allocates the array and sets the transport/device count limit. My UCX code change then sets the number of pairs actually returned.